src/segments-group.js

```diff
@@ -451,29 +448,43 @@ define([
 
           if (newEndTime > nextSegment.startTime) {
             // there is collision
-            newEndTime = Math.min(nextSegment.endTime - nextSegment.minSize, newEndTime);
+            if (nextSegment.endTime - nextSegment.minSize < newEndTime) {
+              newEndTime = nextSegment.endTime - nextSegment.minSize;
+              endLimited = true;
+            }
 
             nextSegment.startTime = newEndTime;
             this._updateSegment(nextSegment);
           }
         }
         else {
-          newEndTime = Math.min(
-            this._sourceGroup.getSource().endTime - this._sourceGroup.getSource().startTime,
-            newEndTime
-          );
+          if (this._sourceGroup.getSource().endTime - this._sourceGroup.getSource().startTime
+              < newEndTime) {
+            newEndTime = this._sourceGroup.getSource().endTime
+                        - this._sourceGroup.getSource().startTime;
+            endLimited = true;
+          }
         }
       }
     }
 
     // Check for minimal size of segment
-    if (newStartTime !== null) {
+    if (newStartTime !== null && newEndTime !== null) {
+      if (newEndTime - newStartTime < segment.minSize) {
+        if (startLimited) {
+          newEndTime = newStartTime + segment.minSize;
+        }
+        else if (endLimited) {
+          newStartTime = newEndTime - segment.minSize;
+        }
+      }
+    }
+    else if (newStartTime !== null) {
       if (segment.endTime - newStartTime < segment.minSize) {
         newStartTime = segment.endTime - segment.minSize;
       }
     }
-
-    if (newEndTime !== null) {
+    else if (newEndTime !== null) {
```

Comment:
You can write it like this
```
else if (newEndTime) {
```
or  you should use "!=" and "==" to work with "null", but I can't think of the reason, you can read about it, for example, here:
http://adripofjavascript.com/blog/drips/equals-equals-null-in-javascript.html

Reply:
I can't write `else if (newEndTime) {` as 0 would be considered faulty (and we don't want that. I tried to use != and == before but the linter doesn't allow it. Should we change it?

Reply:
ah, ok, let's keep it like that than.